app/models/bishop.rb

```diff
@@ -4,4 +4,22 @@ def move_legal?(to_row, to_col)
     col_diff = (col - to_col).abs
     row_diff == col_diff
   end
+
+  def path_to_king
+    king = game.pieces.where.not(is_black: is_black).where(type: 'King').first
+    return nil unless (king.row - row).abs == (king.col - col).abs
```

Comment:
Inside `path_to_king` it seems there's a little bit of duplication where we're redefining again how a piece can move - would it make sense to do something like this?

```ruby
return nil unless valid_move?(king.row, king.col)
```